examples/tutorials/SMPC_Protocols_Explanation.ipynb

```diff
@@ -0,0 +1,390 @@
+{
```

Comment:
<p>For mat mul, it could be good to add that the beaver triples extend to matrices</p><p> </p><p>the "Low Level" components of SecureNN are not necessarily useful unless you explain precisely how they use each other to implement ReLU for example</p><p> </p><p>You mention that division is possible, it is indeed with SecureNN but it is incredibly slow -  you can mention it</p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/4635/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Reply:
<p>Alright, thanks! </p><p>Do you know why specifically the division is slow? </p><p> </p><p>Also the idea about the "Low Level" components was to give more advanced users the possibility to themselves assess whether special computations are possible with the implemented low level algorithms of the protocol. Detailed explanations about how the algorithms work might scare of the less experienced users, how about I could include the graph showing which low level components work together and are based on each other (Fig.3 in the paper). What do you think? </p>

Reply:
<p>&gt; Do you know why specifically the division is slow?</p><p> </p><p>Yeah because it you check the implem, you can see that it makes roughly n call to the comparison, and because the comparison is already a bit expensive, it gets terrible^^</p><p> </p><p>Yeah Fig 3 is helpful I agree! But even for advanced users, we don't expect them to reuse parts of the low level components to build new computations, they should instead rely on the the Relu, Div, and MaxPool I would say</p>

Reply:
Ahh ok because the division is implemented as subtraction and then bitwise comparison and shifting, right?

Ok so I think I'll just shift the low-level parts with the figures to an extra section at the end for users who are more interested and want to have a starting point to understand the implementations and the papers. 👍